2006Scape Server/src/main/java/com/rs2/game/players/PlayerAssistant.java

```diff
@@ -2033,7 +2033,8 @@ public boolean addSkillXP(double amount, int skill) {
 		if (player.tutorialProgress < 36 && player.playerLevel[skill] == 3 && GameConstants.TUTORIAL_ISLAND) {
 			return false;
 		}
-		amount *= GameConstants.XP_RATE;
+
+		amount *= GameConstants.XP_RATE * player.getXPRate();
```

Comment:
Should we not be checking for `VARIABLE XP RATE` here?
If using variable XP rate, ignore the base XP rate, if using variable XP rate, use the players set XP rate.
Set the players default start XP rate to be whatever the base XP rate is on account creation?

Something like this for this line (haven't tested):
```suggestion
		if (GameConstants.VARIABLE_XP_RATE) {
			amount *= player.getXPRate();
		} else {
			amount *= GameConstants.XP_RATE;
		}
```

Reply:
I thought about that but it technically works just the same because if it's on a server with variable XP Rates disabled then the players XP rate will be 1 anyway, the current way just makes it cleaner by one if statement and means people can set a global XP rate ontop of the players specific XP rates if they'd like to

Reply:
Currently, if the servers BASE XP rate is 5, and then the players VARIABLE XP rate is 10, they would be getting 50x XP though

Reply:
Yeah that would be the server owners choice if they change the global XP while enabling VARIABLE_XP_RATES, I feel like it makes more sense to have the global XP rate apply to everything/everyone that way if a server wanted to make the minimum XP rate x5 while still allowing players to multiply that with the VARIABLE_XP_RATES they could

Reply:
Also if you brought a save over from a VARIABLE XP rate server to a non, they would still be getting the boosted XP.
Just doesn't make sense IMO to be multiplying these 2 together.

Reply:
But then wouldn't it be better to make VARIABLE XP rate an array, so you could just specify the rates you wanted instead of having a multiplier on top of all the variable xp rates?

Reply:
Making it an array would probably make it cleaner anyway to be honest, I was just trying to get it done because of how long ago I polled the second world & I was thinking of making the Players XP rate only load if VARIABLE_XP_RATES is set that way the saves would have cross compatibility with servers that have VARIABLE_XP_RATES disabled